ledger-api/grpc-definitions/com/daml/ledger/api/v1/active_contracts_service.proto

```diff
@@ -12,6 +12,7 @@ option java_outer_classname = "ActiveContractsServiceOuterClass";
 option java_package = "com.daml.ledger.api.v1";
 option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 
+// TODO clarification how activeness and in-transfer relate
```

Comment:
@andreaslochbihler-da : how does per-domain pruning of transfer-events work? do you prune both transfer-in and transfer-out event once both are present?

Reply:
> @andreaslochbihler-da : how does per-domain pruning of transfer-events work? do you prune both transfer-in and transfer-out event once both are present?

We don't have pruning implemented for transfers.

Reply:
> I'd expect that in-transfer contracts are included in the ACS response as a create event marked as in-flight (with all the meta-data that one gets on the transfer-out) event.

Yes, we need a way to query the active transfers. I recommend to NOT call this a `create` event because it doesn't come from a creation. I'd much more prefer calling it what it is, namely a transfer-in event (or maybe `Enter`). The crucial difference is that there may be only one create event for each contract id ever, while there may be multiple transfer-in events for the same contract.

I think it makes sense to ship the active transfers targeting a domain as part of an ACS snapshot for the domain.

Reply:
I just realize that the request here has an implicit offset as the as_at of the ACS.
Current plans do not involve creating a merged-domain-offset, so what would work here at all is a per domain ACS (symmetrically to the per domain transaction streams).
So I would suggest to add the domain_id here as well to the request.
This would also imply that a contract should be not part of the ACS if it was transferred out, no?

But I was also thinking about the semantics of the command interpretation on the ledger api: would it be acceptable to keep this limited to create and archive events (so current implementation of contract store would not change in the ledger API server)?
@meiersi-da @andreaslochbihler-da

Reply:
> So I would suggest to add the domain_id here as well to the request.

Yes. That was my understanding of what is needed for CN: strictly per-domain streams and snapshots.

> This would also imply that a contract should be not part of the ACS if it was transferred out, no?

Correct. For the period between transfer-out and transfer-in, the contract should be listed somehow as "in transit". I'd list the contract on the target domain because that's how the implementation currently works (and the contract can only be transferred-in on the target domain anyway).

Reply:
> would it be acceptable to keep this limited to create and archive events

What does "this" refer to? We're discussing there the ACS snapshot, which does not contain archived events, does it? So archived doesn't make sense to me here. I think we certainly need an endpoint for dealing with "contracts in transit", e.g., to figure out what the transfer-out ID is that one wants to use in a transfer-in submission.